www/front_src/src/Extensions/Listing/index.tsx

```diff
@@ -0,0 +1,468 @@
+  );
+
+  React.useEffect(() => {
+    const types = find(propEq('name', 'types'), getAppliedFilterCriteriasAtom);
+    const statuses = find(
+      propEq('name', 'statuses'),
+      getAppliedFilterCriteriasAtom,
+    );
+
+    if (types && types.value) {
+      const typesValues = types.value as Array<SelectEntry>;
+      setModulesActive(!!find(propEq('id', 'MODULE'), typesValues));
+      setWidgetsActive(!!find(propEq('id', 'WIDGET'), typesValues));
+    }
+
+    sendExtensionsRequests({
+      endpoint: buildExtensionEndPoint({
+        action: 'list',
+        criteriaStatus: statuses,
+      }),
+    }).then(({ status, result }) => {
+      if (status) {
+        setExtension(result as Extensions);
+      } else {
+        showErrorMessage(result as string);
+      }
+    });
+  }, [getAppliedFilterCriteriasAtom]);
+
+  const getEntitiesByKeyAndVersionParam = (
+    param,
+    equals,
+    key,
+  ): Array<EntityType> => {
+    const resArray: Array<EntityType> = [];
+    if (extensions) {
+      for (let i = 0; i < extensions[key].entities.length; i += 1) {
+        const entity = extensions[key].entities[i];
+        if (entity.version[param] === equals) {
+          resArray.push({
+            id: entity.id,
+            type: key,
+          });
+        }
+      }
+    }
+
+    return resArray;
+  };
+
+  const getAllEntitiesByVersionParam = (param, equals): Array<EntityType> => {
+    if (
+      (!modulesActive && !widgetsActive) ||
+      (modulesActive && widgetsActive)
+    ) {
+      return [
+        ...getEntitiesByKeyAndVersionParam(param, equals, 'module'),
+        ...getEntitiesByKeyAndVersionParam(param, equals, 'widget'),
+      ];
+    }
+    if (modulesActive) {
+      return [...getEntitiesByKeyAndVersionParam(param, equals, 'module')];
+    }
+    // inverted because of inverse logic for switches on/off false/true
+
+    return [...getEntitiesByKeyAndVersionParam(param, equals, 'widget')];
+  };
+
+  const updateAllEntities = (): void => {
+    const entities = getAllEntitiesByVersionParam('outdated', true);
+    if (entities.length <= 0 || !entities) {
+      return;
+    }
+    entities.forEach((entity) => {
+      updateById(entity.id, entity.type);
+    });
+  };
+
+  const installAllEntities = (): void => {
+    const entities = getAllEntitiesByVersionParam('installed', false);
+    if (entities.length <= 0 || !entities) {
+      return;
+    }
+    entities.forEach((entity) => {
+      installById(entity.id, entity.type);
+    });
+  };
+
+  const updateById = (id: string, type: string): void => {
+    setExtensionsUpdatingStatusByIds(id, true);
+    sendUpdateOrInstallExtensionRequests({
+      endpoint: buildEndPoint({
+        action: 'update',
+        id,
+        type,
+      }),
+    })
+      .then(({ status, result }) => {
+        if (!status) {
+          showErrorMessage(result.message as string);
+        } else {
+          showSuccessMessage('successful Update');
```

Comment:
```suggestion
          showSuccessMessage('Update succeeded');
```

You should use a translatedLabel here